src/ray/core_worker/task_manager.cc

```diff
@@ -28,6 +28,26 @@ const int64_t kTaskFailureThrottlingThreshold = 50;
 // Throttle task failure logs to once this interval.
 const int64_t kTaskFailureLoggingFrequencyMillis = 5000;
 
+TaskStatusCounter::TaskStatusCounter() {
+  // TODO(ekl): this isn't actually resetting OpenCensus properly
+  for (int i = 0; i < rpc::TaskStatus_ARRAYSIZE; i++) {
+    ray::stats::STATS_tasks.Record(0, rpc::TaskStatus_Name(i));
+  }
+}
+
+void TaskStatusCounter::Swap(rpc::TaskStatus old_status, rpc::TaskStatus new_status) {
```

Comment:
Nit. Should this class more generic for other resource (e.g., actor)?

Reply:
Not sure--- we can probably address once we add actor stats?

Reply:
Sgtm!